adaptive_n_back/experiment.js

```diff
@@ -200,6 +234,9 @@ for (var b = 0; b < num_blocks; b++) {
 		};
 		adaptive_n_back_experiment.push(test_block)
 	}
+	if ($.inArray(b,[4,7,15]) != -1) {
```

Comment:
I don't actually like the idea of randomly inserting attention checks. I've put them at the end of blocks, or groups of trials in this code so they are unexpected to the user (they don't know when the end of a block is coming) but don't interfere with task performance (by distracting them in the middle of a block). Right now there are no clear signifiers of when this should happen.

In the future we could look at the block's "trial_id", which I'll use to label "task relevant" blocks like "stim" or "feedback" but not text blocks like "trial_start" or "rest". In principle, we could look at the transition from blocks where trial_id is defined to blocks where it isn't and insert attention checks there.

But that is neither here nor there. For now, what would you prefer in terms of experiment variables? I have three options:
1) Define one credit_var that integrates over all possible conditions that I (the experiment designer) assert should be passed to get credit
2) Split up the credit conditions into attention checks and whatever else. Return booleans for both (indicating passed or failed). This allows someone to include attention checks for analytic purposes, but potentially not use them for credit/blacklisting purposes.
3) Have an attention_check_percent variable, which is a numeric 0-1 variable that allows the user to decide a particular criterion for rejection outside of the experiment. This allows different users to keep attention checks but use them more liberally/conservatively by altering _something_ about battery definition in the way we plan to do for all bonus variables. That seems a lot of work (for us) outside of the experiment deifnition since this would have to be done for every experiment with attention checks.


Side point - I'm working with attention-check percents (meaning if someone passes 2/3 they passed 66%). Because there are so few attention checks in every paradigm, does it make sense to instead juts use raw numbers (passed 2 attention checks flat out). The "criteria" is dependent on the number of attention checks regardless, so it's not  like the percentage brings everything into the same space...